packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -294,7 +294,7 @@ class _DevFSHttpWriter {
     } catch (e) {
       if (retry < kMaxRetries) {
         printTrace('Retrying writing "$deviceUri" to DevFS due to error: $e');
-        _scheduleWrite(deviceUri, content, progressReporter, retry + 1);
+        await _scheduleWrite(deviceUri, content, progressReporter, retry + 1);
```

Comment:
It looks like above (line 268), we call this in a synchronous method with no `.then()` because we're relying on the `progressReporter` to notify the caller when it does its thing.  Thus, we should probably not wait here and just ignore the linter warning.